textproc/cfonts/Portfile

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cargo  1.0
+
+github.setup        DominikWilkowski cfonts 1.0.2rust v
+revision            0
+
+categories          textproc
+license             GPL-3+
+maintainers         {DominikWilkowski @DominikWilkowski}
+
+description         Sexy ANSI fonts for the console
+long_description    This is a silly little command line tool for sexy fonts in the console. \
+                    Give your cli some love.
+homepage            https://github.com/dominikwilkowski/cfonts
+
+checksums           ${distfiles} \
+                    rmd160  6f5bedcb0e1cb54faf9efb2b4e34dbe6105152ea \
+                    sha256  33c14dda907c4f3c046a40644c8856f6debb87b58ee6fbaab2b8d7af14ce8b6e \
+                    size    3312272
+
+post-extract {
+    cd rust/
```

Comment:
You can delete this `post-extract` block.  I think what you want here is `build.dir`, which lets you specify to MacPorts the working directory where the build commands should be executed:

```
build.dir        ${worksrcpath}/rust
```

Reply:
Normally Cargo ports expect a `cargo.crates` section generated off of `Cargo.lock`.  You can use something like `cargo2port` to help you do that.  If not, you can remove the `--frozen` flag so that the port will build dynamically (though MacPorts has a preference for explicitly specified dependencies.  To build without `--frozen`, you can do:

```
build.pre_args-delete       --frozen
```

Reply:
fixed both